stock_picking_customer_ref/views/stock_picking_view.xml

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+        <record model="ir.ui.view" id="vpicktree_inh_customerinfo">
+            <field name="name">vpicktree.inh.customerinfo</field>
+            <field name="model">stock.picking</field>
+            <field name="inherit_id" ref="stock.vpicktree" />
+            <field name="arch" type="xml">
+                <field name="partner_id" position="after">
+                    <field name="client_order_ref" />
+                </field>
+            </field>
+        </record>
+
+        <record model="ir.ui.view" id="view_picking_form_inh_customerinfo">
+            <field name="name">view.picking.form.inh.customerinfo</field>
+            <field name="model">stock.picking</field>
+            <field name="inherit_id" ref="stock.view_picking_form" />
+            <field name="arch" type="xml">
+                <field name="group_id" position="after">
+                    <field name="client_order_ref"
+                           attrs="{'invisible':
+                           [('client_order_ref', '=', False)]}" />
```

Comment:
@Swapnesh-SerpentCS You're right. It's better to show the field explicitly because if it's hidden, the user does not know the value - even void - of that field. @bizzappdev